code/Controllers/CMSMain.php

```diff
@@ -2184,9 +2184,10 @@ public function duplicate(HTTPRequest $request): HTTPResponse
         }
 
         if (($id = $this->urlParams['ID']) && is_numeric($id)) {
+            $modelClass = $this->getModelClass();
             /** @var DataObject&Hierarchy $record */
-            $record = DataObject::get($this->getModelClass())->byID($id);
-            if ($record && !$record->canCreate(null, ['Parent' => $record->Parent()])) {
+            $record = DataObject::get($modelClass)->byID($id);
+            if ($record && !$record->canCreate(null, ['Parent' => $record->getParent() ?? $modelClass::create()])) {
```

Comment:
The magic `Parent()` method returns an empty record if it failed to fetch one from the db, but `getParent()` returns `null` - so we have to account for that change.